ludwig/schema/hyperopt/search_algorithm.py

```diff
@@ -0,0 +1,57 @@
+from dataclasses import field
+from typing import Dict
+
+from marshmallow import fields, ValidationError
+from marshmallow_dataclass import dataclass
+
+from ludwig.hyperopt.registry import search_algorithm_registry  # Double-check this implicit import.
+from ludwig.schema import utils as schema_utils
+
+
+@dataclass
+class BaseSearchAlgorithmConfig(schema_utils.BaseMarshmallowConfig):
+    import ludwig.hyperopt.search_algos
+
+    """Basic search algorithm settings."""
+
+    type: str = schema_utils.StringOptions(
+        options=list(search_algorithm_registry.keys()), default="hyperopt", allow_none=False
+    )
+
+
+def SearchAlgorithmDataclassField(description: str = "", default: Dict = {"type": "variant_generator"}):
+    allow_none = False
```

Comment:
May be cleaner to put this as an optional argument with a default if we want to allow the option for None, otherwise, I think it may be better to just put `None` in the return below. WDYT?

Reply:
Good catch, will move `allow_none` into the return. I don't think it should be configurable to a user of the field, though.